odoo/addons/base/models/ir_model.py

```diff
@@ -2210,8 +2210,16 @@ def _module_data_uninstall(self, modules_to_remove):
             model = self.pool.get(ir_field.model)
             if model is not None:
                 field = model._fields.get(ir_field.name)
-                if field is not None:
-                    field.prefetch = False
+                if field is not None and field.prefetch:
```

Comment:
Read with diff of Strategy1

I think it is the safest strategy for stable. It doesn't solve unknown recomputations but at least it solves the problem for shared fields and doesn't create any new corner case